### ISSUE TITLE
[Issue] Help developers recover from capacity errors on free accounts

### ISSUE THREAD

**Issue by savannahostrowski** [Azure/azure-dev#468]
![image](https://user-images.githubusercontent.com/8949415/184409196-4295b4c6-5741-41fe-8170-40b6a5ec6a50.png)

Steps to repro:
1. Create a free account 
2. Pick a template (I believe Mandy used a node template)
3. Choose East US 2 as region (apparently Central US region worked)

**Comment by Austinauth** [comment]
@savannahostrowski a few clarifying questions:

- Is the first step _Create a free account_ something that the user has done prior to this error in the CLI?
- Is a _free account_ the same as a regular Azure account you can create using the Azure web UI or is it something more specific?

**Comment by savannahostrowski** [comment]
1. Yes + you'll need to auth with the free account
2. Free account with $200 credits link -  https://azure.microsoft.com/en-us/free/

**Comment by savannahostrowski** [comment]
Moved this to November!

**Comment by savannahostrowski** [comment]
This is part of planned work we have for Gallium to improve error messaging.

**Comment by savannahostrowski** [comment]
Important for: https://github.com/Azure-Samples/azure-search-openai-demo/pull/252

**Comment by rajeshkamal5050** [comment]
@ChrisHMSFT do you know the status and coverage of the preflight API coverage on the RP's?

**Comment by ChrisHMSFT** [comment]
Yes, all except the capacity check is supported today. You can use: https://learn.microsoft.com/en-us/rest/api/resources/deployments/validate?view=rest-resources-2021-04-01

**Comment by rajeshkamal5050** [comment]
Based on comment here https://github.com/Azure/azure-dev/pull/4329#issuecomment-2492583470 moving to backlog. Depends on fix from bicep/preflight team fixing nested resources handling. @hemarina can you add the corresponding dependency issue here for tracking?
